components/Form/index.js

```diff
@@ -0,0 +1,122 @@
+import styled from "styled-components";
+
+export default function Form({ mutate }) {
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const { title, content } = Object.fromEntries(formData);
+
+    const response = await fetch("/api/upload", {
+      method: "POST",
+      body: formData,
+    });
+
+    if (!response.ok) {
+      console.error("Whoopsie");
+      return;
+    }
+
+    try {
+      const { height, width, url } = await response.json();
+
+      const newPost = {
+        title,
+        image: { height, width, url },
+        content,
+      };
+
+      await fetch("/api/posts", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newPost),
+      });
+
+      mutate();
```

Comment:
why should the frontend create a second fetch? This could be done in the api route after the image was uploaded.

Reply:
This was my initial approach as well, to be honest.

After consulting with @onemanwenttomow I decided to go with the approach we currently see.

The reasoning behind this being that our `/api/upload/index.js` Route would then exclusively be used for uploading images to Cloudinary.
And our `/api/posts/index.js` Route would only be used to interact with our own database.

Moreover, when fetching the `posts` API Route from our `upload` API Route, I encountered issues related to resolving the URL path that I could only solve by prefixing `/api/posts` with `${request.headers.origin}`.
While this worked, I was worried that this might cause issues in a deployed version of the application.